File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/model/Station.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.model;

import androidx.annotation.NonNull;

import java.io.Serializable;

public class Station implements Serializable {

    private String country;

    private String id;
    private String title;
    private double lat;
    private double lon;
    private String ds100;
    private String photoUrl;
    private String photographer;
    private String photographerUrl;
    private String license;
    private String licenseUrl;
    private boolean active;
    private boolean outdated;
    private long photoId = -1;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLon() {
        return lon;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Station)) return false;

        Station station = (Station) o;

        return id.equals(station.id);
    }

    @Override
    @NonNull
    public String toString() {
        return "Bahnhof [id=" + id + ", title=" + title + ", lat=" + lat + ", lon=" + lon + ", DS100=" + ds100 + "]";
    }

    public String getDs100() {
        return ds100;
    }

    public void setDs100(String ds100) {
        this.ds100 = ds100;
    }

    public void setPhotoUrl(String photoUrl) {
        this.photoUrl = photoUrl;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public void setPhotographer(String photographer) {
        this.photographer = photographer;
    }

    public String getPhotographer() {
        return photographer;
    }

    public void setPhotographerUrl(String photographerUrl) {
        this.photographerUrl = photographerUrl;
    }

    public String getPhotographerUrl() {
        return photographerUrl;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public String getLicense() {
        return license;
    }

    public String getLicenseUrl() {
        return licenseUrl;
    }

    public void setLicenseUrl(String licenseUrl) {
        this.licenseUrl = licenseUrl;
    }

    public boolean hasPhoto() {
        return photoUrl != null;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public boolean isOutdated() {
        return outdated;
    }

    public void setOutdated(boolean outdated) {
        this.outdated = outdated;
    }

    public void setPhotoId(long photoId) {
        this.photoId = photoId;
    }

    public long getPhotoId() {
        return this.photoId;
    }

}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/model/Station.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.model

import java.io.Serializable

data class Station @JvmOverloads constructor(
    val country: String,
    val id: String,
    val title: String,
    val lat: Double,
    val lon: Double,
    val ds100: String? = null,
    val photoUrl: String? = null,
    val photographer: String? = null,
    val photographerUrl: String? = null,
    val license: String? = null,
    val licenseUrl: String? = null,
    val active: Boolean = false,
    val outdated: Boolean = false,
    val photoId: Long?,
) : Serializable {

    fun hasPhoto(): Boolean {
        return photoUrl != null
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Station

        if (country != other.country) return false
        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        var result = country.hashCode()
        result = 31 * result + id.hashCode()
        return result
    }

}